### ISSUE TITLE
Issue in annotationFinalForStrVCTVRE.py

### ISSUE THREAD

**Issue by looxon93** [andrewSharo/StrVCTVRE#7]
HI, I wanted to try StrVCTVRE tool, however I am running into the error. Even test_StrVCTVRE.py returns an error: 

```
python3.9 test_StrVCTVRE.py 

reading VCF...

formatting VCF data...

identifying exonic deletions and duplications...

scoring exonic deletions and duplications...

Traceback (most recent call last):
  File "/opt/StrVCTVRE/test_StrVCTVRE.py", line 152, in <module>
    annotationFinalForStrVCTVRE.annotateSVs(os.path.join(td,'svsForAnnotation.csv'), os.path.join(td,'svsAnnotated.csv'), args.phylopPath, td)
  File "/opt/StrVCTVRE/annotationFinalForStrVCTVRE.py", line 137, in annotateSVs
    x = np.asarray(x)
ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (507,) + inhomogeneous part.
```

Do you maybe know what is causing this?

**Comment by andrewSharo** [comment]
Hi Luka,

Thanks for submitting this issue. I've not seen this error before. It occurs in the part of the script where StrVCTVRE annotates variants with the phyloP file hg38.phyloP100way.bw. Can you double-check that this file looks correct? It should be 9.2 GB. 

If you are able to send me the vcf/bed file that you're trying to score with StrVCTVRE, I'm happy to try to reproduce this error on my system. Even if you can send just a subset of your file that causes the error, that would be helpful. 

Andrew

**Comment by looxon93** [comment]
Hi Andrew, thank you for the quick reply! I did checksum of phyloP file and everything seems ok, so file is downloaded correctly. Even test isn't working. Can you download this docker image with command: 

`docker pull genieusgenomics/strvctvre1.8:0`

Run the image interactively with command: 

`docker run -it genieusgenomics/strvctvre1.8:0 `

And there just run test command: 

`python3.9 test_StrVCTVRE.py -p data/hg38.phyloP100way.bw`

These three commands should be enough for you to reproduce the issue, if not I will share some test files. But since even test isn't working, I suppose it must do to something with reference. Let me know if you found anything. 

Thanks for all your help! 
Luka.

**Comment by andrewSharo** [comment]
Hi Luka,

I read your first post too quickly and missed that it was the test script that was returning an error. Thanks for reiterating that point. Your docker image was very easy to use. I've been able to recreate the error, and there were several issues, mostly caused by changes introduced in new version of libraries. As you may have noticed, StrVCTVRE was developed in python 2, so compatibility with python 3 is not perfect. I have fixed several errors and released a new version of StrVCTVRE (1.9). However, the test_StrVCTVRE.py program still does not run in your docker image, as it gives this error:

```
reading VCF...


formatting VCF data...


identifying exonic deletions and duplications...


scoring exonic deletions and duplications...

Traceback (most recent call last):
  File "/opt/StrVCTVRE/test_StrVCTVRE.py", line 166, in <module>
    rf = load('data/rfTrainedAllChromsExcept'+chrm+'.joblib')
  File "/usr/local/lib/python3.9/site-packages/joblib/numpy_pickle.py", line 658, in load
    obj = _unpickle(fobj, filename, mmap_mode)
  File "/usr/local/lib/python3.9/site-packages/joblib/numpy_pickle.py", line 577, in _unpickle
    obj = unpickler.load()
  File "/usr/local/lib/python3.9/pickle.py", line 1210, in load
    dispatch[key[0]](self)
  File "/usr/local/lib/python3.9/pickle.py", line 1526, in load_global
    klass = self.find_class(module, name)
  File "/usr/local/lib/python3.9/pickle.py", line 1577, in find_class
    __import__(module, level=0)
ModuleNotFoundError: No module named 'sklearn.ensemble.forest'
```

I believe this is caused by an issue where joblib is looking for an sklearn module that it cannot find because sklearn has updated its naming of sklearn.ensemble.forest to sklearn.ensemble._forest. One option would be for you to downgrade sklearn to 0.22 or earlier (the docker appears to use 1.3.0). There may be a more elegant solution, but I could not find it. Another option would be for me to regenerate these joblib random forest files, but that will take some time. 

I'm really glad that you're interested in using StrVCTVRE. I'm happy to continue to help you get it running. I do want to mention a few additional considerations:

Have you considered using the provided environment_py2.7.yml file to create a python environment that is guaranteed to work? I recognize that python 2 is no longer supported, and I am working on getting a python 3 version of StrVCTVRE available, but it will take some time.  

If you have a small number of variants to score, you could also try our web server (https://strvctvre.berkeley.edu/annotate.html).

Thanks again for raising these issues!

Andrew

**Comment by looxon93** [comment]
Hi @andrewSharo , I finally managed tool to work. Installing older libraries just created new problems, so I installed conda environment you provided in the repo here. Since I managed to make the tool to work inside a docker container, I am providing a Dockerfile (extension txt is added so it can be uploaded) so you can easily containerize your tool, it might be useful for others. It isn't the cleanest one, but it works. 
[Dockerfile.txt](https://github.com/andrewSharo/StrVCTVRE/files/12054089/Dockerfile.txt)

Thanks again for your help.
